src/bundle/Core/Command/VirtualFieldDuplicateFixCommand.php

```diff
@@ -0,0 +1,272 @@
+
+                // Wait, if needed, before moving to next iteration
+                usleep($sleep * 1000);
+            } while ($batchSize === count($attributes));
+
+            $style->success(sprintf(
+                'Operation successful. Removed total of %d duplicates. [Debug %s]',
+                $totalDeleted,
+                $stopwatch->stop('total')
+            ));
+        } catch (Exception $exception) {
+            $style->error($exception->getMessage());
+
+            return Command::FAILURE;
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function getDuplicatedAttributeTotalCount(
+        SymfonyStyle $style,
+        Stopwatch $stopwatch
+    ): int {
+        $stopwatch->start('total_count', 'sql');
+        $query = $this->connection->createQueryBuilder()
+            ->select('COUNT(a.id) as instances')
+            ->groupBy('version', 'contentclassattribute_id', 'contentobject_id', 'language_id')
+            ->from('ezcontentobject_attribute', 'a')
+            ->having('instances > 1');
+
+        $count = $query->execute()->rowCount();
+
+        if ($count > 0) {
+            $style->warning(
+                sprintf(
+                    'Found %d of affected attributes. [Debug: %s]',
+                    $count,
+                    $stopwatch->stop('total_count')
+                )
+            );
+        }
+
+        return $count;
+    }
+
+    /**
+     * @phpstan-return array<array{
+     *     version: int,
+     *     contentclassattribute_id: int,
+     *     contentobject_id: int,
+     *     language_id: int,
+     * }>
+     */
+    private function getDuplicatedAttributesBatch(int $batchSize): array
+    {
+        $query = $this->connection->createQueryBuilder();
+
+        $query
+            ->select('version', 'contentclassattribute_id', 'contentobject_id', 'language_id')
+            ->groupBy('version', 'contentclassattribute_id', 'contentobject_id', 'language_id')
+            ->from('ezcontentobject_attribute')
+            ->having('COUNT(id) > 1')
+            ->setFirstResult(0)
+            ->setMaxResults($batchSize);
+
+        return $query->execute()->fetchAllAssociative();
+    }
+
+    /**
+     * @phpstan-param array{
+     *     version: int,
+     *     contentclassattribute_id: int,
+     *     contentobject_id: int,
+     *     language_id: int
+     * } $attribute
+     *
+     * @return int[]
+     */
+    private function getDuplicatedAttributeIds(array $attribute): array
+    {
+        $query = $this->connection->createQueryBuilder();
+
+        $query
+            ->select('id')
+            ->from('ezcontentobject_attribute')
+            ->where('version = :version')
+            ->andWhere('contentclassattribute_id = :contentclassattribute_id')
+            ->andWhere('contentobject_id = :contentobject_id')
+            ->andWhere('language_id = :language_id')
+            ->orderBy('id', 'ASC')
+            ->setFirstResult(0);
```

Comment:
You're discarding the first row later.

You can skip it then.

```suggestion
            ->orderBy('id', 'ASC')
            ->setFirstResult(1);
```

Also, are you sure that the first row, with the lowest ID, should be kept? Not the highest one?